### Задача 2
Розглянемо дві різні умовні акції X і Y. Для кожної з них за допомогою статистичних досліджень були розраховані відносні частоти очікуваних прибутків залежно від п’яти станів економічного середовища: значне піднесення (ЗП), незначне піднесення (НЗП), стагнація (С), незначний спад (НЗС), значний спад (ЗС). Ці дані занесені відповідно в останній рядок і останній стовпчик таблиці 1. Залежно від економічного середовища прогнозовані норми очікуваних прибутків у відсотках задані відповідно у першому рядку та першому стовпчику таблиці 1. У клітинках на перетині X і Y проставлені ймовірності появи норм відсотків значень (x, y).


|Y   |X=10 |X=5 |X=2 |X=1 |X=-5|P(y)|
|----|-----|----|----|----|----|----|
|20  |0.01 |0.03|0.02|0.03|0.01|0.1 |
|10  |0.03 |0.09|0.06|0.09|0.03|0.3 |
|2   |0.02 |0.06|0.04|0.06|0.02|0.2 |
|-2  |0.03 |0.09|0.06|0.09|0.03|0.3 |
|-8  |0.01 |0.03|0.02|0.03|0.01|0.1 |
|P(x)|0.1  |0.3 |0.2 |0.3 |0.1 |1   |

Необхідно:  
1) Обслити очікувану норму прибутку по кожній з акцій;
2) Оцінити ступінь ризику обох акцій за допомогою середньоквадратичного відхилення;
3) Проаналізувати наявність зв’язку між акціями X та Y за допомогою коефіцієнтів коваріації та кореляції.



### Розв'язання
Обчислимо математичні характеристики виборки за допомогою програми.
Використаємо формули:

$$M(X)= \sum_{i=0}^{4}X_i P_i \,\,\,\,\,\,\,\,\,\,\,\,\,M(Y)= \sum_{i=0}^{4}Y_i P_i\,\,\,\,\,\,\,\,\,\,M(X,Y)=\sum_{i=0}^{4} \sum_{j=0}^{4} Y_iX_jP_{ij}$$

$$D(X)=\sum_{i=1}^{3}X_i^2 P_i - M(X)^2 \,\,\,\,\,\,\,D(Y)=\sum_{i=1}^{3}Y_i^2 P_i - M(Y)^2 \,\,\,\,\,\,\,\,\,\, \sigma(X)=\sqrt{D(X)}  \,\,\,\,\,\,\,\,\,\,\,\,\,\,\,\, \sigma(Y)=\sqrt{D(Y)}$$

$$cov(X,Y)=M(X,Y)-M(X)M(Y) \,\,\,\,\,\,\,\,\,\,\,\,\, r_{xy}=\frac{cov(X,Y)}{\sqrt{D(X)D(Y)}}  $$

In [33]:
x=[10,5,2,1,-5]
y=[20,10,2,-2,-8]
px=[0.1,0.3,0.2,0.3,0.1]
py=[0.1,0.3,0.2,0.3,0.1]
pxy=[[0.01,0.03,0.02,0.03,0.01],
     [0.03,0.09,0.06,0.09,0.03],
     [0.02,0.06,0.04,0.06,0.02],
     [0.03,0.09,0.06,0.09,0.03],
     [0.01,0.03,0.02,0.03,0.01]]
n=5

print("X  ",end="")
for i in range(n-1):
    print("     ",x[i],end="")
print("     ",x[n-1])
print("Px ",end="")
for i in range(n-1):
    print("    ",px[i],end="")
print("   ",px[n-1])
print("")
print("Y  ",end="")
for i in range(n-1):
    print("     ",y[i],end="")
print("    ",y[n-1])
print("Py   ",end="")
for i in range(n-1):
    print("   ",py[i],end="")
print("   ",py[n-1])
print("")
for j in range(n):
    for i in range(n-1):
        print('P(',j,',',i,')=','{0:7.3f}  '.format(pxy[j][i]),end="") 
    print('P(',j,',',4,')=','  {0:7.3f}  '.format(pxy[j][n-1]))
    
mxy=0  # Математичні очікування
mx=0
my=0
for i in range(n):
    mx+=x[i]*px[i]
    my+=y[i]*py[i]
    for j in range(n):
        mxy+=y[i]*x[j]*pxy[i][j]
print('Математичне очікування')
print('Mx={0:8.5f}'.format(mx),'My={0:8.5f}'.format(my),'Mxy={0:8.5f}'.format(mxy))

dx=0 #Дисперсії
dy=0
for i in range(n):
    dx+=x[i]*x[i]*px[i]
    dy+=y[i]*y[i]*py[i]
dx=dx-mx*mx
dy=dy-my*my
print('Дисперсія')
print('DX={0:8.5f}'.format(dx),"   DY={0:8.5f}".format(dy))

# Розрахунок середньоквадратичного відхилення
print("Середньоквадратичне відхилення")
print('Xaverage={0:8.5f}'.format(dx**0.5),"   Yavarage={0:8.5f}".format(dy**0.5))

# Розрахунок коваріації
cov=mxy-mx*my
print("Коваріація  cov(xy)={0:8.5f}".format(cov))

# Розрахунок кореляції
rxy=cov/(dx*dy)**0.5
print("Кореляція  Rxy={0:8.5f}".format(rxy))

X        10      5      2      1      -5
Px      0.1     0.3     0.2     0.3    0.1

Y        20      10      2      -2     -8
Py       0.1    0.3    0.2    0.3    0.1

P( 0 , 0 )=   0.010  P( 0 , 1 )=   0.030  P( 0 , 2 )=   0.020  P( 0 , 3 )=   0.030  P( 0 , 4 )=     0.010  
P( 1 , 0 )=   0.030  P( 1 , 1 )=   0.090  P( 1 , 2 )=   0.060  P( 1 , 3 )=   0.090  P( 1 , 4 )=     0.030  
P( 2 , 0 )=   0.020  P( 2 , 1 )=   0.060  P( 2 , 2 )=   0.040  P( 2 , 3 )=   0.060  P( 2 , 4 )=     0.020  
P( 3 , 0 )=   0.030  P( 3 , 1 )=   0.090  P( 3 , 2 )=   0.060  P( 3 , 3 )=   0.090  P( 3 , 4 )=     0.030  
P( 4 , 0 )=   0.010  P( 4 , 1 )=   0.030  P( 4 , 2 )=   0.020  P( 4 , 3 )=   0.030  P( 4 , 4 )=     0.010  
Математичне очікування
Mx= 2.70000 My= 4.00000 Mxy=10.80000
Дисперсія
DX=13.81000    DY=62.40000
Середньоквадратичне відхилення
Xaverage= 3.71618    Yavarage= 7.89937
Коваріація  cov(xy)=-0.00000
Кореляція  Rxy=-0.00000


### Відповідь
Очікувана норма прибутку по акції X =2,7%,  по акції Y=4 %   
Очікуваний ризик по акції X$\approx$3.7%, по акції Y$\approx$7.9%  
Акції є інвестиційно непривабливими.    
Коефіцієнт кореляції між акціями дорівнює 0 що свідчить про те, що акції незалежні.